### PR TITLE
Add helptext, default logic to `ipfs stats`

### PR DESCRIPTION
**Pull request by RichardLitt**
I've added: Synopsis, shortDescription, and longDescription for stats and stats bw.
I also added default(false) for poll and default("1s") for interval, and modified the logic to make it fit.

Part of #2484 and #2647.

License: MIT
Signed-off-by: Richard Littauer <richard.littauer@gmail.com>

### CODE REVIEW

**Comment by Kubuxu** [comment]
LGMT

IIRC `ipfs stats bw` was quite broken some time ago, I don't know if it got fixed but it is out of the scope in this case.

**Comment by RichardLitt** [comment]
Updated synopsis to match `man git`. 

Yeah, `ipfs stats` seems pretty neglected.

**Comment by RichardLitt** [comment]
Fixed spacing.

**Comment by Kubuxu** [comment]
LGTM but you have binary files in your PR.

**Comment by RichardLitt** [comment]
Fixed.

File: core/commands/stat.go [hunk 1]
```diff
@@ -18,8 +18,10 @@ import (
 
 var StatsCmd = &cmds.Command{
 	Helptext: cmds.HelpText{
-		Tagline:          "Query IPFS statistics.",
-		ShortDescription: ``,
+		Tagline:          "Query ipfs statistics.",
+		Synopsis:         "ipfs stats <command>",
+		ShortDescription: `'ipfs stats' is a set of commands to help look at statistics for your ipfs node.`,
+		LongDescription: `'ipfs stats' is a set of commands to help look at statistics for your ipfs node.`,
```
Comment (whyrusleeping) [inline core/commands/stat.go]: I think we can probably add some logic that sets the long description to the short description if unset

Comment (whyrusleeping) [inline core/commands/stat.go]: (not that you have to change anything here, just saying that might be a nice feature)

File: core/commands/stat.go [hunk 2]
```diff
@@ -116,14 +150,14 @@ var statBwCmd = &cmds.Command{
 					totals := nd.Reporter.GetBandwidthTotals()
 					out <- &totals
 				}
-				if !doPoll {
-					return
-				}
-				select {
```
Comment (whyrusleeping) [inline core/commands/stat.go]: this change isnt right. The existing logic says to halt execution after the first print if we arent in polling mode.

Comment (Kubuxu) [inline core/commands/stat.go]: I think it is what will happen over here.

If `doPoll` is false it will go straight to the `return statement in line 160.

Comment (whyrusleeping) [inline core/commands/stat.go]: but the change is that now if `doPoll` is true, it still exits. It just waits, and then exits

File: core/commands/stat.go (hunk 1 above)
Comment (RichardLitt) [inline core/commands/stat.go]: Opened https://github.com/ipfs/go-ipfs/issues/2691.

File: core/commands/stat.go (hunk 2 above)
Comment (RichardLitt) [inline core/commands/stat.go]: Reverted.

**Comment by Kubuxu** [comment]
LGTM

File: core/commands/stat.go [hunk 3]
```diff
@@ -78,20 +115,17 @@ var statBwCmd = &cmds.Command{
...
 			return
 		}
-		if found {
-			v, err := time.ParseDuration(timeS)
-			if err != nil {
-				res.SetError(err, cmds.ErrNormal)
-				return
-			}
-			interval = v
+		v, err := time.ParseDuration(timeS)
```
Comment (whyrusleeping) [inline core/commands/stat.go]: `interval, err := ...`

**Pull request merged**